python/deltalake/table.py

```diff
@@ -210,17 +210,14 @@ def metadata(self) -> Metadata:
         """
         return self._metadata
 
-    def vacuum(self, retention_hours: int, dry_run: bool = True) -> List[str]:
+    def vacuum(self, retention_hours: Optional[int] = None, dry_run: bool = True) -> List[str]:
         """
         Run the Vacuum command on the Delta Table: list and delete files no longer referenced by the Delta table and are older than the retention threshold.
 
         :param retention_hours: the retention threshold in hours
         :param dry_run: when activated, list only the files, delete otherwise
         :return: the list of files no longer referenced by the Delta Table and are older than the retention threshold.
         """
-        if retention_hours < 0:
-            raise ValueError("The retention periods should be positive.")
-
```

Comment:
I included this check in the Python binding to match https://github.com/delta-io/delta/blob/master/core/src/main/scala/org/apache/spark/sql/delta/commands/VacuumCommand.scala#L58, but it would be better to move it inside the `VACUUM` rust method.

Reply:
Oh ok, the reason why I removed this check is that 1) I could not figure out how check on optional in python and 2) the call would fail anyway, because if you add the lesser value that default it will fail anyways, even if 0. 

But yeah, I'll re-add this check on rust side then